rfcs/63-nursery.md

```diff
@@ -0,0 +1,168 @@
+  - [ ] can we get there without requiring a full rewrite?
+- [ ] are there any dependencies to third-party plugins?
+- [ ] can we reduce our dependency tree by standardizing with `bevy` itself or rewriting small dependencies?
+- [ ] does the crate contain any features that we have explicitly relegated to the third-party ecosystem?
+- [ ] can niche functionality be split out into ecosystem crates that builds on the foundation established by the new core engine crate?
+- [ ] does the crate need to be renamed / re-versioned as part of the integration process?
+
+Crates **do not have to be complete** to be merged into Bevy itself: merely polished and clearly useful.
+
+The PR to merge the crate into Bevy itself should be very simple, and contain only the bare minimum change set required to enable interoperability.
+Further integration work should be done in follow-up PRs.
+If you are an interested community member, raise issues and submit fixes on the nursery repository **before** the crate is being upstreamed.
+
+Once the upstreaming PR is merged, the nursery crate's repo will be archived and any still-relevant issues will be migrated to the main Bevy repo.
+
+## Drawbacks
+
+1. Clearly picks winners within the crate ecosystem.
+   1. These winners typically already exist, but it's hard for beginners to discover that critical information.
+   2. Competing crates may end up being merged as part of the nursery process.
+2. Might increase the scope of the core engine.
+   1. Nursery crates should never expand the planned scope of the engine; merely avoid reimplementation of planned work.
+   2. Feature flags, template games and plugin sets help ensure users only compile and ship the features they care about.
+3. Upstreaming additional parts of the engine without adding more maintainers will increase the workload and may slow down progress.
+   1. This can be addressed independently.
+   2. PR contributors and nursery contributors should be treated equally.
+4. The authors of upstreamed crates will lose some creative control over their crates.
+   1. They will still be able to make PRs to the core engine, and their opinion will be weighed heavily.
+   2. Nursery crate authors may eventually become maintainers in their own right, in no small part due to their work creating and refining the nursery crate.
+   3. This is ultimately no worse than a hard fork or reimplementation, which is the direct alternative.
+
+## Rationale and alternatives
+
+### Why should we upstream crates?
+
+In other modular ecosystems, it's very common to have critical functionality outside of the core project.
+This is unfortunate because:
+
+1. It's very challenging for beginners to discover these essential tools.
+2. These tools cannot readily be used in official documentation.
+3. These tools may suddenly go unmaintained.
+4. Work may be divided across competitors (including work done by the core).
+5. Integration with the core may be poor, or suddenly break.
+
```

Comment:
I would like to add that when depending on external plugins, especially more than one, it is often very painful to track Bevy main (even pointing to a specific commit), and a benefit of upstreaming plugins is that CI is guaranteed to pass.

Reply:
I would not expect nursery crates to track main tightly (they'll live in seperate crates), so I don't think this is actually an advantage.

This is the right call IMO to avoid increasing burden and friction too heavily.

Reply:
At the very least, a nursery might give the community a set of crates to more vigilantly keep up to date with main. However, as Alice said, nursery crates won't be in lockstep with bevy commits.

Reply:
Yeh i understand, I just meant once they're fully upstreamed, not during the nursery phase. 

Nevermind, it's not really important.